modules/stereo/src/matching.cpp

```diff
@@ -0,0 +1,338 @@
+            int widthDisp2;
+            widthDisp2 = widthDisp;
+            widthDisp -= 1;
+            for (int i = 0; i <= widthDisp; i++)
+            {
+                if (c[(iw + i * search_region) * widthDisp2 + i] < mini)
+                {
+                    mini3 = mini2;
+                    mini2 = mini;
+                    mini = c[(iw + i * search_region) * widthDisp2 + i];
+                    index = i;
+                }
+                else if (c[(iw + i * search_region) * widthDisp2 + i] < mini2)
+                {
+                    mini3 = mini2;
+                    mini2 = c[(iw + i * search_region) * widthDisp2 + i];
+                }
+                else if (c[(iw + i * search_region) * widthDisp2 + i] < mini3)
+                {
+                    mini3 = c[(iw + i * search_region) * widthDisp2 + i];
+                }
+            }
+            if (mini3 / mini <= confidenceCheck)
+                return index;
+            return -1;
+        }
+        //!Interpolate in order to obtain better results
+        double Matching::symetricVInterpolation(int *c, int iwjp, int widthDisp, int winDisp,const int search_region)
+        {
+            if (winDisp == 0 || winDisp == widthDisp - 1)
+                return winDisp;
+            double m2m1, m3m1, m3, m2, m1;
+            m2 = c[(iwjp + (winDisp - 1) * search_region) * widthDisp + winDisp - 1];
+            m3 = c[(iwjp + (winDisp + 1) * search_region)* widthDisp + winDisp + 1];
+            m1 = c[(iwjp + winDisp * search_region) * widthDisp + winDisp];
+            m2m1 = m2 - m1;
+            m3m1 = m3 - m1;
+            if (m2m1 == 0 || m3m1 == 0) return winDisp;
+            double p;
+            p = 0;
+            if (m2 > m3)
+            {
+                p = (0.5 - 0.25 * ((m3m1 * m3m1) / (m2m1 * m2m1) + (m3m1 / m2m1)));
+            }
+            else
+            {
+                p = -1 * (0.5 - 0.25 * ((m2m1 * m2m1) / (m3m1 * m3m1) + (m2m1 / m3m1)));
+            }
+            if (p >= -0.5 && p <= 0.5)
+                p = winDisp + p;
+            return p;
+        }
+        //!Generate the hamming LUT
+        void Matching::hammingLut()
+        {
+            for (int i = 0; i <= 65536; i++)
+            {
+                int dist = 0;
+                int j = i;
+                //we number the bits from our number
+                while (j)
+                {
+                    dist = dist + 1;
+                    j = j & (j - 1);
+                }
+                hamLut[i] = dist;
+            }
+        }
+        //!remove small regions that have an area smaller than t, we fill the region with the average of the good pixels around it
+        void Matching::smallRegionRemoval(const Mat &currentMap, int t, Mat &out)
+        {
+            CV_Assert(currentMap.cols == out.cols);
+            CV_Assert(currentMap.rows == out.rows);
+            CV_Assert(t > 0);
+            memset(pus, 0, 2000000 * sizeof(pus[0]));
```

Comment:
Just a imagine how disastrous will it be if you'll decide to shrink the array a bit and forget to fix this memset.

BTW, for a static array `sizeof` returns the length of the whole array in bytes.